### DISCUSSION TITLE
I love this project so far

### DISCUSSION THREAD

**Discussion by goffy59** [mrrfv/open-android-backup#46] [Q&A] [answered]
I want to say thanks for the great work thus far. I was wondering if you had plans to implement text/sms backup. That would seriously help. My brother passed away a couple years ago and I've still going through his stuff and trying to archive things and now I am finally doing his phones and I would like to save the texts. There is a way to save the texts but it involves using an APP and I don't like that because I don't want it in some proprietary package. I like the way your script handles it. Would he nice to have it simply copy texts into .txt format or whatever people would like I don't care either way. Anyhow, just a question and suggestion and a thanks for your work!

**Comment by mrrfv** [comment]
Thank you for the feedback, and sorry for your loss. The reason #23 didn't see any progress so far is because competing proprietary apps seem to directly access and modify the SMS database, something I couldn't figure out how to do in Flutter so far. A SMS message exporter is entirely doable, although such export wouldn't be restorable. I'll see if anything can be done.

**Comment by mrrfv** [comment, marked as answer]
SMS message exporting has been added in commit https://github.com/mrrfv/linux-android-backup/commit/0b9ce12ed9e1abd51facfab73cca5d2f927e1851 (i.e. latest version). You just have to take a new backup and SMS messages will be included in it. There are a few things to note though:

- These messages cannot be restored to a new device
- They are viewable by opening the backup archive (the file that this script creates on your computer)
- Attachments (images, videos) probably aren't saved, only text
- You should use LibreOffice or Microsoft Excel to view the exported messages, although a standard text editor also works.
- If the address is 0, it means that the message was sent, not received.

FYI I have a donation link in my GitHub profile if you'd like to support me financially. No pressure though.
